vm/systemSmartContracts/auction.go

```
@@ -137,95 +137,127 @@ func (s *stakingAuctionSC) unJail(args *vmcommon.ContractCallInput) vmcommon.Ret
 	totalUnJailPrice := big.NewInt(0).Mul(config.UnJailPrice, big.NewInt(int64(len(args.Arguments))))
 
 	if totalUnJailPrice.Cmp(args.CallValue) != 0 {
+		s.eei.AddReturnMessage("insufficient funds sent for unJail")
 		return vmcommon.UserError
 	}
 
 	err := s.eei.UseGas(s.gasCost.MetaChainSystemSCsCost.UnJail * uint64(len(args.Arguments)))
 	if err != nil {
+		s.eei.AddReturnMessage("insufficient gas limit")
 		return vmcommon.OutOfGas
 	}
 
 	for _, argument := range args.Arguments {
-		vmOutput, _ := s.executeOnStakingSC([]byte("unJail@" + hex.EncodeToString(argument)))
-		if vmOutput != nil && vmOutput.ReturnCode == vmcommon.OutOfGas {
+		vmOutput, err := s.executeOnStakingSC([]byte("unJail@" + hex.EncodeToString(argument)))
+		if err != nil {
+			s.eei.AddReturnMessage(err.Error())
+			continue
+		}
+
+		if vmOutput.ReturnCode == vmcommon.OutOfGas {
```

this error is not returned anymore

---

the condition can be deleted

---

done.